SwiftyBluetooth/Source/SBError.swift

```diff
@@ -50,7 +50,9 @@ public enum SBError: Error {
     case scanningEndedUnexpectedly
     case operationTimedOut(operation: SBOperation)
     case invalidPeripheral
+    case peripheralFailedToConnectWithError(error: Error)
```

Comment:
Nope, the various SBError you see here only exist for cases where an error happens where no CoreBluetooth error exist to describe what's going on. For example, the the timeout error is something CoreBluetooth does not have because the delegate calls do not timeout. This is the reason why all the Error in the callback definitions are casted as "Error" instead of "SBError" even though often times only SBErrors are returned by these functions, it allows for the library to send back an error from CoreBluetooth directly when they happen, for example when there is a connection error.